src/systemdhook.c

```diff
@@ -58,6 +61,117 @@ DEFINE_CLEANUP_FUNC(yajl_val, yajl_tree_free)
 
 #define CGROUP_ROOT "/sys/fs/cgroup"
 
+static int makepath(char *dir, mode_t mode)
+{
+    if (!dir) {
+	errno = EINVAL;
+	return -1;
+    }
+
+    if (strlen(dir) == 1 && dir[0] == '/')
+	return 0;
+
+    makepath(dirname(strdupa(dir)), mode);
+
+    return mkdir(dir, mode);
+}
+
+static int bind_mount(const char *src, const char *dest, const char *opt) {
+	if (mount(src, dest, "bind", MS_BIND|MS_REC, NULL) == -1) {
+		pr_perror("Failed to mount %s on %s", src, dest);
+		return -1;
+	}
+	if (mount(src, dest, "bind", MS_REMOUNT|MS_BIND|MS_REC, opt) == -1) {
+		pr_perror("Failed to remount %s", dest);
+		return -1;
+	}
+	return 0;
+}
+
+/* error callback */
+static int parser_errcb(struct libmnt_table *tb __attribute__ ((__unused__)),
+			const char *filename, int line)
+{
+	pr_perror("%s: parse error at line %d", filename, line);
+	return 0;
+}
+
+static struct libmnt_table *parse_tabfile(const char *path)
+{
+	int rc;
+	struct libmnt_table *tb = mnt_new_table();
+
+	if (!tb) {
+		pr_perror("failed to initialize libmount table");
+		return NULL;
+	}
+
+	mnt_table_set_parser_errcb(tb, parser_errcb);
+
+	rc = mnt_table_parse_file(tb, path);
+
+	if (rc) {
+		mnt_free_table(tb);
```

Comment:
Could probably use defer style cleanup like in rest of the file using gcc attributes. See https://github.com/mrunalp/hooks/pull/22/files for an example with libmnt_table.